### PR TITLE
resolves #84 Displaying QR inline with carousel

### LINKED ISSUE #84: Displaying QR inline with carousel
The purpose of this feature is to enable the displaying of QR inline with a carousel instead of inline with line break.
It can be useful for chatbot with lower width or for modal chatbot when there is a lot of quick replies.

Concept :
- It should be an option and not activated by default in order to not impact other users of tock-react-kit
- The carousel has to be generic and customizable like the card carousel

Preview :
image

### PR DESCRIPTION
**Pull request by correi-f**
resolves #84 Displaying QR inline with carousel

### CODE REVIEW

**Review by francoisno — APPROVED** [review]
Ok, thx for the storybook addition too.
It would be nice to add the accessibility attributes and overrides consistently with !88 don't you think @correi-f ?

**Comment by correi-f** [comment]
> Ok, thx for the storybook addition too. It would be nice to add the accessibility attributes and overrides consistently with !88 don't you think @correi-f ?

Yes, I agree but I prefer not to do this change until the accessibility is merged. Can accessibility be prioritized over this change ?

**Review by elebescond — APPROVED** [review]
Ok (run eslint --fix to fix formatting problems)

**Comment by correi-f** [comment]
> Ok (run eslint --fix to fix formatting problems)

Done

File: src/components/InlineQuickReplyList/InlineQuickReplyList.tsx [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+        {items.map((child, index) =>
+          React.cloneElement(
+            <QuickReply
+              key={`${child.label}-${index}`}
+              onClick={onItemClick.bind(null, child)}
+              {...child}
+            />,
+            { ref: ref.items[index] },
+            undefined,
+          ),
```
Comment (Fabilin) [inline src/components/InlineQuickReplyList/InlineQuickReplyList.tsx:145]: Given that parents cannot get direct refs to the `QuickReply` here, couldn't this be simplified to
```suggestion
            <QuickReply
              key={`${child.label}-${index}`}
              onClick={onItemClick.bind(null, child)}
              {...child}
              ref={ref.items[index]}
            />
```

File: src/components/InlineQuickReplyList/hooks/useButtonRefs.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  useEffect(() => {
+    const initialRefs = Array.from(Array(count)).map(() =>
+      createRef<HTMLButtonElement>(),
+    );
+    setRefs(initialRefs);
+  }, []);
+
+  useEffect(() => {
+    setRefs((refs) =>
+      Array.from(Array(count)).map((_, i) => refs[i] || createRef()),
```
Comment (Fabilin) [inline src/components/InlineQuickReplyList/hooks/useButtonRefs.ts:17]:
```suggestion
      Array.from({length: count}, (_, i) => refs[i] || createRef()),
```

File: src/components/InlineQuickReplyList/InlineQuickReplyList.tsx (hunk 1 above)
Comment (correi-f) [reply src/components/InlineQuickReplyList/InlineQuickReplyList.tsx:145]: I already tried this implementation bud it did not work, I do not know really why unless I did a mistake somewhere else.

Comment (correi-f) [reply src/components/InlineQuickReplyList/InlineQuickReplyList.tsx]: I did a mistake last time for sure, it seems to work. I merged your implementation.

**Review by Fabilin — APPROVED** [review]

**Comment by francoisno** [comment]
@correi-f could you rebase on _master_ and resolve the conflicts please? (I'm not sure of doing it correctly myself)

**Pull request merged**